NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Search.swift

```diff
@@ -31,51 +31,48 @@ extension NASourceAnimeUnity {
             private var _results: [AnimeLink]?
             var title: String
             var stringa = ""
-        var searchRequestUrl: URL {
-            parent.endpointURL.appendingPathComponent("inc/livesearch.php")
-        }
+            var returned = true
             weak var delegate: ContentProviderDelegate?
             
             func links(on page: Int) -> [AnyLink] {
                 page == 0 ? _results?.map { .anime($0) } ?? [] : []
             }
             
             func more() {
-                let params: Parameters = ["query": title]
-                let url = "https://animeunity.it/anime.php?c=archive"
-                _ = AF.request(url, method: .post, parameters: params).responseData {response in
-                    do {
-                        let str_correct = response.debugDescription
-//                        str_correct = self.stringa
-                        let bowl = try SwiftSoup.parse(str_correct)
-                        let entries = try bowl.select("div.row>div.col-lg-4")
-                        self._results = try entries.compactMap {
-                            entry -> AnimeLink? in
-//                            do {
-                                let animeTitle = try entry.select("div>h6.card-title>b"/*"h6.card-title>b"*/)
-                                let title = try animeTitle.text()
-                                if title.isEmpty { return nil }
-                                let animeLinkPath = try entry.select("div>a")
-                                let url = try animeLinkPath.attr("href")
-                                let trimmed = url.trimmingCharacters(in: .whitespacesAndNewlines)
-
-                                let animeArtworkPath = try entry.select("img").attr("src")
-                                guard let animeUrl = URL(string: "https://animeunity.it/" + trimmed),
-                                    let coverImage = URL(string: animeArtworkPath)
-                                else {
-                                    return nil
-                                }
-                            return AnimeLink(
-                                    title: title,
-                                    link: animeUrl,
-                                    image: coverImage,
-                                    source: self.parent
-                                )
+                if(self.returned) {
+                    self.returned = false
+                    let params: Parameters = ["query": title]
+                    let url = "https://animeunity.it/anime.php?c=archive"
+                    _ = AF.request(url, method: .post, parameters: params).responseData {response in
```

Comment:
* Although this code works right now, using `self.parent.browserSession.request` is preferred over `AF.request`.
* Do not hold a strong reference to self inside this escaping closure. Use `[weak self]` in case references to this `SearchAgent` have been removed. Delegate methods should not be called when no longer referencing this `SearchAgent`.
* In addition, make sure you apply the middlewares since they are used to detect and resolve numerous user identity challenges.
* Hold a reference to the `DataRequest` object returned by the `.request()` method. Cancels the `DataRequest` when `SearchAgent` is deinitializing.

Reference the code below: https://github.com/SuperMarcus/NineAnimator/blob/3aff9a02d670048608ba6864bb867c5f9a931a61/NineAnimator/Models/Anime%20Source/animeunity.it/AnimeUnity%2BSearch.swift#L57-L63